added translation for state a components memory (#586)
reactjs/es.reactjs.org: beta/src/content/learn/state-a-components-memory.md

---

> title: "El Estado: Como memoria de un componente"

Reviewer: @Vicenciomf1 ¿Por qué ha desaparecido la preposición "de" del título? En mi humilde opinión, sin ella, la frase deja de tener sentido. Debería ser "El estado: La memoria de un componente".

> title: "El Estado: Como memoria de un componente"

Reviewer 2: Hola @bonavida! Tienes toda la razón, sinceramente no sé cómo esa pertenencia se me fue, generalmente reviso hartas veces antes de hacer alguna sugerencia (probablemente sea debido a que hice dos comentarios, y copié y pegué el anterior para usar los _suggested changes_ de Github en el nuevo comentario).

Actualmente tengo una lista con muchos _typos_ de este tipo, pienso presentarlos todos en un PR, probablemente veas los cambios en máximo una semana. ¡Muchas gracias por tu comentario!
P. D: No te preocupes por comentar la rama mergeada, sinceramente no lo hubiese notado sin aquél, así que muchas gracias.

}, {
  name: 'Eternal Presence',
  artist: 'John Woodrow Wilson',

>   description: 'Esta enorme flor plateada (75 pies o 23 m) se encuentra en Buenos Aires. Está diseñado para moverse, cerrando sus pétalos por la tarde o cuando soplan fuertes vientos y abriéndolos por la mañana.',

Reviewer 2: ```suggestion
  description: 'Wilson era conocido por su preocupación por la igualdad y la justicia social, así como por las cualidades esenciales y espirituales de la humanidad. Esta enorme pieza de bronce (de 7 pies o 2,13 metros) representa lo que él describió como "una presencia negra simbólica impregnada de un sentido de humanidad universal".',
```
Razón: Este par key-value está desfasado, la descripción corresponde a la key anterior (la description de la Floraris Genérica).

}, {
  name: 'Terracotta Army',
  artist: 'Artista Desconocido',
  description: 'El Ejército de terracota es una colección de esculturas de terracota que representan los ejércitos de Qin Shi Huang, el primer emperador de China. El ejército constaba de más de 8.000 soldados, 130 carros con 520 caballos y 150 caballos de caballería.',
  url: 'https://i.imgur.com/HMFmH6m.jpg',
  alt: '12 esculturas de terracota de guerreros solemnes, cada uno con una expresión facial y una armadura únicas.'
}, {
  name: 'Lunar Landscape',
  artist: 'Louise Nevelson',
  description: 'Nevelson era conocida por recoger objetos de los escombros de la ciudad de Nueva York, que luego ensamblaría en construcciones monumentales. En este, usó partes dispares como un poste de la cama, un alfiler de malabares y un fragmento de asiento, clavándolos y pegándolos en cajas que reflejan la influencia de la abstracción geométrica del espacio y la forma del cubismo.',
  url: 'https://i.imgur.com/rN7hY6om.jpg',
  alt: 'Una escultura negra mate donde los elementos individuales son inicialmente indistinguibles.'
}, {
  name: 'Aureole',
  artist: 'Ranjani Shettar',

>   description: 'SShettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad y una "fina síntesis de materiales inverosímiles.',

Reviewer 2: ```suggestion
  description: 'Shettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad, y una "fina síntesis de materiales inverosímiles.',
```
Razón: Corregido un typo (Doble S), y una coma para mejorar cohesión (mantiene el sujeto, que es la descripción de las obras).

</Sandpack>

El controlador de eventos `handleClick` está actualizando una variable local, `index`. Pero dos cosas impiden que ese cambio sea visible:

1. **Las variables locales no persisten entre renderizaciones.** Cuando React renderiza este componente por segunda vez, lo hace desde cero, no considera ningún cambio en las variables locales.
2. **Los cambios en las variables locales no activarán renderizaciones.** React no se da cuenta de que necesita renderizar el componente nuevamente con los nuevos datos.

Para actualizar un componente con datos nuevos, deben pasar las siguientes dos cosas:

1. **Conservar** los datos entre renderizaciones.
2. **Provocar** que React renderice el componente con nuevos datos (re-renderizado).

El Hook de [`useState`](/apis/react/useState) ofrece dos cosas:

1. Una **variable de estado** para mantener los datos entre renderizados.

> 2. Una **función que setea el estado** para actualizar la variable y provocar que React renderice el componente nuevamente.

Reviewer 2: El único contenido ya publicado que causaría conflicto para dejarlo como _settear_, sería [este capítulo](https://beta.es.reactjs.org/learn/referencing-values-with-refs)

> 2. Una **función que setea el estado** para actualizar la variable y provocar que React renderice el componente nuevamente.

Author: Realmente a mí también me gusta más el termino de setear. De todos modo quedo atenta a lo que ustedes decidan qué es lo mejor.

> ## Agregando una variable de estado {/*adding-a-state-variable*/}

Reviewer 2: ```suggestion
## Agregar una variable de estado {/*adding-a-state-variable*/}
```
La [traducción directa del gerundio](https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#el-gerundio) con función sustantiva en los títulos podría perder significado, por eso es recomendado utilizar un infinitivo (Agregando->Agregar).

Para agregar una variable de estado, debemos importar el `useState` de React al inicio del archivo:

```js
import { useState } from 'react';
```

Luego, reemplazamos esta línea:

```js
let index = 0;
```

con

```js
const [index, setIndex] = useState(0);
```

`index` es una variable de estado y `setIndex` es la función que setea el estado.

> > La sintáxis de `[` y `]` se le conoce cómo [destructuración del array](https://javascript.info/destructuring-assignment) y es lo que permite leer los valores de un array. El array retornado por `useState` siempre contará con exactamente dos valores.

Reviewer 2: ```suggestion
> A la sintaxis de `[` y `]` se le conoce cómo [desestructuración de un array](https://javascript.info/destructuring-assignment) y permite leer valores de un array. El array retornado por `useState` siempre contará con exactamente dos valores.
```

}, {
  name: 'Eternal Presence',
  artist: 'John Woodrow Wilson',

>   description: 'Esta enorme flor plateada (75 pies o 23 m) se encuentra en Buenos Aires. Está diseñado para moverse, cerrando sus pétalos por la tarde o cuando soplan fuertes vientos y abriéndolos por la mañana.',

Reviewer 2: ```suggestion
  description: 'Wilson era conocido por su preocupación por la igualdad y la justicia social, así como por las cualidades esenciales y espirituales de la humanidad. Esta enorme pieza de bronce (de 7 pies o 2,13 metros) representa lo que él describió como "una presencia negra simbólica impregnada de un sentido de humanidad universal".',
```
Razón: Este par key-value está desfasado, la descripción corresponde a la key anterior (la description de la Floraris Genérica).

}, {
  name: 'Terracotta Army',
  artist: 'Artista Desconocido',
  description: 'El Ejército de terracota es una colección de esculturas de terracota que representan los ejércitos de Qin Shi Huang, el primer emperador de China. El ejército constaba de más de 8.000 soldados, 130 carros con 520 caballos y 150 caballos de caballería.',
  url: 'https://i.imgur.com/HMFmH6m.jpg',
  alt: '12 esculturas de terracota de guerreros solemnes, cada uno con una expresión facial y una armadura únicas.'
}, {
  name: 'Lunar Landscape',
  artist: 'Louise Nevelson',
  description: 'Nevelson era conocida por recoger objetos de los escombros de la ciudad de Nueva York, que luego ensamblaría en construcciones monumentales. En este, usó partes dispares como un poste de la cama, un alfiler de malabares y un fragmento de asiento, clavándolos y pegándolos en cajas que reflejan la influencia de la abstracción geométrica del espacio y la forma del cubismo.',
  url: 'https://i.imgur.com/rN7hY6om.jpg',
  alt: 'Una escultura negra mate donde los elementos individuales son inicialmente indistinguibles.'
}, {
  name: 'Aureole',
  artist: 'Ranjani Shettar',

>   description: 'SShettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad y una "fina síntesis de materiales inverosímiles.',

Reviewer 2: ```suggestion
  description: 'Shettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad, y una "fina síntesis de materiales inverosímiles.',
```
Razón: Corregido un typo (Doble S), y una coma para mejorar cohesión (mantiene el sujeto, que es la descripción de las obras).

</Sandpack>

### Conoce tu primer Hook {/*meet-your-first-hook*/}

En React, `useState`, así como cualquier otra función que empiece con "`use`", se le conoce como Hook.

> Los *Hooks* son funciones especiales que sólo están disponibles mientras React está [renderizando](/learn/render-and-commit#step-1-trigger-a-render) (algo que veremos con más detalle en la página siguiente). Los Hooks permiten "engancharnos" a diferentes funciones de React.

Reviewer 2: ```suggestion
Los *Hooks* son funciones especiales que sólo están disponibles mientras React está [renderizando](/learn/render-and-commit#step-1-trigger-a-render) (algo que veremos con más detalle en la página siguiente). Los Hooks permiten "engancharnos" a diferentes características de React.
```
Razón: Al usar ya el término "funciones" antes, podría tender a la confusión el usar tal palabra otra vez, si usamos "funcionalidades", o "características" por ejemplo, podemos mantener el sentido y a la vez evitar la confusión para la gente nueva en React, además, así mantenemos la consistencia con la traducción de la palabra en la siguiente línea "El estado es solo una de esas características,".

El estado es solo una de esas características, pero conoceremos los otros Hooks más adelante.

<Pitfall>

> **Las funciones-Hook que empiecen con `use` deben ser solo llamadas en el nivel superior de los componentes o [en tus propios Hooks.](/learn/reusing-logic-with-custom-hooks)** No podemos usar Hooks dentro de condicionales, bucles, u otras funciones anidadas. Los Hooks son funciones, pero es útil pensar en ellos como declaraciones incondicionales sobre las necesidades de nuestro componente. Las funciones de React se "usan" en la parte superior del componente de manera similar a cómo se "importan" módulos en la parte superior de un archivo.

Reviewer 2: ```suggestion
**Las funciones-Hook que empiecen con `use` deben ser solo llamadas en el nivel superior de los componentes o [en tus propios Hooks.](/learn/reusing-logic-with-custom-hooks)** No podemos usar Hooks dentro de condicionales, bucles, u otras funciones anidadas. Los Hooks son funciones, pero es útil pensar en ellos como declaraciones independientes de las necesidades de nuestro componente. Las funciones de React se "usan" en la parte superior del componente de manera similar a cómo se "importan" módulos en la parte superior de un archivo.
```
Está un poco difícil la traducción, pero cambiando un poco las palabras podemos mantener el significado de la oración original, al usar incondicional->independiente podemos mantener el sentido de que no debemos cambiar el lugar en el que ejecutamos el hook dependiendo de nuestras necesidades, como declarar estados debido a condicionales, o dentro de mapeos, try...catch, declaraciones de funciones, etc, ya que si hacemos eso, entonces React perderá el orden y/o cantidad en su array de arrays [getter, setter] de nuestros estados (si no están en top level, el call stack no te queda de la misma manera)

</Pitfall>

### Anatomía del `useState` {/*anatomy-of-usestate*/}

Cuando llamamos al [`useState`](/apis/react/useState), le estamos diciendo a React que debe recordar algo:

```js
const [index, setIndex] = useState(0);
```

En este caso, queremos que React recuerde el `index`.

> > La convención es nombrar estas dos variables como `const [algo, establecerAlgo]`. Podemos nombrarlo como queramos, pero mantener las convenciones hacen que las cosas sean más fáciles de entender en todos los proyectos.

Reviewer 2: ```suggestion
> La convención es nombrar estas dos variables como `const [algo, setAlgo]`. Podemos nombrarlo como queramos, pero mantener las convenciones hacen que las cosas sean más fáciles de entender en todos los proyectos.
```
Razón: La convención que se usa comúnmente es [estado, setEstado] (o [estado, setEstado]), al usar "establecerAlgo", estamos perdiendo la convención de usar el prefijo de "set" para hacer alusión al setter (porque devuelve un array con un [getter, setter]).

El único argumento para `useState` es el **valor inicial** de su variable de estado. En este ejemplo, el valor inicial de `index` se establece en `0` con `useState(0)`.

> Cada vez que el componente se renderiza, el `useState` devuelve una matriz que contiene dos valores:

Reviewer 2: ```suggestion
Cada vez que el componente se renderiza, el `useState` devuelve un *array* que contiene dos valores:
```
Razón: Así mantenemos la consistencia con los usos anteriores y posteriores en este capítulo, y los capítulos anteriores en la documentación.

> 1. La **variable de estado** (`index`) con el valor que almacenó.

Reviewer 2: ```suggestion
1. La **variable de estado** (`index`) con el valor que almacenaste.
```
Razón: Solemos usar la [segunda persona singular "tú"](https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#usted-t%C3%BA-y-vos) para mantener el tono informal de la documentación original.

2. La **función que establece el estado** (`setIndex`) que puede actualizar la variable de estado y alertar a React para que renderice el componente nuevamente.

Así es como sucede en acción:

```js
const [index, setIndex] = useState(0);
```

> 1. **Tu componente se renderiza la primera vez.** Debido a que pasamos `0` a `useState` como valor inicial para `index`, esto devolverá `[0, setIndex]`. React recuerda que `0` es el último valor de estado.

Reviewer 2: ```suggestion
1. **Tu componente se renderiza por primera vez.** Debido a que pasamos `0` a `useState` como valor inicial para `index`, esto devolverá `[0, setIndex]`. React recuerda que `0` es el último valor de estado.
```

2. **Actualizas el estado.** Cuando un usuario hace clic en el botón, llama a `setIndex(index + 1)`. `index` es `0`, por lo tanto es `setIndex(1)`. Esto le dice a React que recuerde que `index` es `1` ahora y ejecuta otro renderizado.
3. **El componente se renderiza por segunda vez.** React todavía ve `useState(0)`, pero debido a que React *recuerda* que estableció `index` en `1`, devuelve `[1, setIndex]` en su lugar.
4. ¡Y así sucesivamente!

> ## Colocando múltiples variables de estado a un componente {/*giving-a-component-multiple-state-variables*/}

Reviewer 2: ```suggestion
## Colocar múltiples variables de estado a un componente {/*giving-a-component-multiple-state-variables*/}
```
La [traducción directa del gerundio](https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#el-gerundio) con función sustantiva en los títulos podría perder significado, por eso es recomendado utilizar un infinitivo (Colocando->Colocar, dar, asignar).

  return (
    <>
      <button onClick={handleNextClick}>
        Siguiente
      </button>
      <h2>
        <i>{sculpture.name} </i> 

>         by {sculpture.artist}

Reviewer 2: 1. Falta traducir la preposición
   * Frase: "by"
   * Sugerencia: "por"
   * Razón: Así mantenemos la traducción de strings al español de manera consistente (la idea es mantenerlo en inglés, o en español, pero no ambas).

>         by {sculpture.artist}

Reviewer 2: ```suggestion
        por {sculpture.artist}
```
Razón: Así mantenemos la traducción de strings al español de manera consistente (la idea es mantenerlo en inglés, o en español, pero no ambas).

}, {
  name: 'Eternal Presence',
  artist: 'John Woodrow Wilson',

>   description: 'Esta enorme flor plateada (75 pies o 23 m) se encuentra en Buenos Aires. Está diseñado para moverse, cerrando sus pétalos por la tarde o cuando soplan fuertes vientos y abriéndolos por la mañana.',

Reviewer 2: ```suggestion
  description: 'Wilson era conocido por su preocupación por la igualdad y la justicia social, así como por las cualidades esenciales y espirituales de la humanidad. Esta enorme pieza de bronce (de 7 pies o 2,13 metros) representa lo que él describió como "una presencia negra simbólica impregnada de un sentido de humanidad universal".',
```
Razón: Este par key-value está desfasado, la descripción corresponde a la key anterior (la description de la Floraris Genérica).

}, {
  name: 'Terracotta Army',
  artist: 'Artista Desconocido',
  description: 'El Ejército de terracota es una colección de esculturas de terracota que representan los ejércitos de Qin Shi Huang, el primer emperador de China. El ejército constaba de más de 8.000 soldados, 130 carros con 520 caballos y 150 caballos de caballería.',
  url: 'https://i.imgur.com/HMFmH6m.jpg',
  alt: '12 esculturas de terracota de guerreros solemnes, cada uno con una expresión facial y una armadura únicas.'
}, {
  name: 'Lunar Landscape',
  artist: 'Louise Nevelson',
  description: 'Nevelson era conocida por recoger objetos de los escombros de la ciudad de Nueva York, que luego ensamblaría en construcciones monumentales. En este, usó partes dispares como un poste de la cama, un alfiler de malabares y un fragmento de asiento, clavándolos y pegándolos en cajas que reflejan la influencia de la abstracción geométrica del espacio y la forma del cubismo.',
  url: 'https://i.imgur.com/rN7hY6om.jpg',
  alt: 'Una escultura negra mate donde los elementos individuales son inicialmente indistinguibles.'
}, {
  name: 'Aureole',
  artist: 'Ranjani Shettar',

>   description: 'SShettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad y una "fina síntesis de materiales inverosímiles.',

Reviewer 2: ```suggestion
  description: 'Shettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad, y una "fina síntesis de materiales inverosímiles.',
```
Razón: Corregido un typo (Doble S), y una coma para mejorar cohesión (mantiene el sujeto, que es la descripción de las obras).

</Sandpack>

> Es una buena idea tener múltiples variables de estado si no se encuentran relacionadas entre sí, como `index` y `showMore` en este ejemplo. Pero si encontramso que a menudo cambiamos dos variables de estado juntas, podría ser mejor combinarlas en una sola. Por ejemplo, si tenemos un formulario con muchos campos, es más conveniente tener una única variable de estado que contenga un objeto que una variable de estado por campo. [Eligiendo la estructura del estado](/learn/choosing-the-state-structure) tiene más consejos sobre esto.

Reviewer 2: ```suggestion
Es una buena idea tener múltiples variables de estado si no se encuentran relacionadas entre sí, como `index` y `showMore` en este ejemplo. Pero si encontramso que a menudo cambiamos dos variables de estado juntas, podría ser mejor combinarlas en una sola. Por ejemplo, si tenemos un formulario con muchos campos, es más conveniente tener una única variable de estado que contenga un objeto que una variable de estado por campo. [Elegir la estructura del estado](/learn/choosing-the-state-structure) tiene más consejos sobre esto.
```
La [traducción directa del gerundio](https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#el-gerundio) con función sustantiva en los títulos podría perder significado, por eso es recomendado utilizar un infinitivo (Eligiendo->Elegir).

<DeepDive title="¿Cómo sabe React qué estado devolver?">

> Es posible que hayas notado que la llamada `useState` no recibe ninguna información sobre *a qué* variable de estado se refiere. No hay un "identificador" que se pase a `useState`, entonces, ¿cómo sabe cuál de las variables de estado debería devolver? ¿Se basa en algún tipo de magia para esto? La respuesta es no.

Reviewer 2: * Frase: "Es posible que hayas notado que la llamada `useState` no recibe"
   * Sugerencia: "Es posible que hayas notado que la llamada a `useState` no recibe"

En cambio, para habilitar su sintaxis concisa, los Hooks **se basan en un orden de llamada estable en cada representación del mismo componente.** Esto funciona bien en la práctica porque si seguimos la regla anterior ("solo llamar a los Hooks en el nivel superior"), los Hooks siempre se llamarán en el mismo orden. Además, este [complemento para el linter] (https://www.npmjs.com/package/eslint-plugin-react-hooks) detecta la mayoría de los errores.

> Internamente, React mantiene una matriz de pares de estados para cada componente. También mantiene el índice de par actual, el cual se establece en `0` antes de ser renderizado. Cada vez que llamamos a `useState`, React devuelve el siguiente par de estados e incrementa el índice. Puedes leer más sobre este mecanismo en [React Hooks: No es magia, sólo son Arrays.](https://medium.com/@ryardley/react-hooks-not-magic-just-arrays-cd4f1857236e)

Reviewer 2: 1. Convención de [traducción común](https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#traducciones-comunes)
   * Frase: "Internamente, React mantiene una matriz de pares de estados para cada componente."
   * Sugerencia: "Internamente, React mantiene un _array_ de pares de estados para cada componente."
   * Razón: Así mantenemos la consistencia con los usos anteriores en este capítulo, y los capítulos anteriores en la documentación.

Este ejemplo **no usa React** pero nos da una idea de cómo funciona `useState` internamente:

<Sandpack>

```js index.js active
let componentHooks = [];
let currentHookIndex = 0;

// Cómo funciona useState dentro de React (simplificado).
function useState(initialState) {
  let pair = componentHooks[currentHookIndex];
  if (pair) {
    // Este no es el primer render,
    // entonces el par de estados ya existe.
    // Devuélvelo y prepárate para la próxima llamada del Hook.
    currentHookIndex++;
    return pair;
  }

  // Esta es la primera vez que estamos renderizando,

>   // así que crea un arreglo de dos posiciones y guárdalo.

Reviewer 2: 1. Convención de [traducción común](https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#traducciones-comunes)
   * Frase: "crea un arreglo"
   * Sugerencia: "crea un _array_"
   * Razón: Así mantenemos la consistencia con los usos anteriores y posteriores en este capítulo, y los capítulos anteriores en la documentación.

  }

  let sculpture = sculptureList[index];
  // Este ejemplo no usa React, entonces
  // devuelve un objeto como resultado en lugar de JSX.
  return {
    onNextClick: handleNextClick,
    onMoreClick: handleMoreClick,

>     header: `${sculpture.name} by ${sculpture.artist}`,

Reviewer 2: 1. Falta traducir la preposición
   * Frase: "by"
   * Sugerencia: "por"
   * Razón: Así mantenemos la traducción de strings al español de manera consistente (la idea es mantenerlo en inglés, o en español, pero no ambas).

}, {
  name: 'Eternal Presence',
  artist: 'John Woodrow Wilson',

>   description: 'Esta enorme flor plateada (75 pies o 23 m) se encuentra en Buenos Aires. Está diseñado para moverse, cerrando sus pétalos por la tarde o cuando soplan fuertes vientos y abriéndolos por la mañana.',

Reviewer 2: ```suggestion
  description: 'Wilson era conocido por su preocupación por la igualdad y la justicia social, así como por las cualidades esenciales y espirituales de la humanidad. Esta enorme pieza de bronce (de 7 pies o 2,13 metros) representa lo que él describió como "una presencia negra simbólica impregnada de un sentido de humanidad universal".',
```
Razón: Este par key-value está desfasado, la descripción corresponde a la key anterior (la description de la Floraris Genérica).

}, {
  name: 'Terracotta Army',
  artist: 'Artista Desconocido',
  description: 'El Ejército de terracota es una colección de esculturas de terracota que representan los ejércitos de Qin Shi Huang, el primer emperador de China. El ejército constaba de más de 8.000 soldados, 130 carros con 520 caballos y 150 caballos de caballería.',
  url: 'https://i.imgur.com/HMFmH6m.jpg',
  alt: '12 esculturas de terracota de guerreros solemnes, cada uno con una expresión facial y una armadura únicas.'
}, {
  name: 'Lunar Landscape',
  artist: 'Louise Nevelson',
  description: 'Nevelson era conocida por recoger objetos de los escombros de la ciudad de Nueva York, que luego ensamblaría en construcciones monumentales. En este, usó partes dispares como un poste de la cama, un alfiler de malabares y un fragmento de asiento, clavándolos y pegándolos en cajas que reflejan la influencia de la abstracción geométrica del espacio y la forma del cubismo.',
  url: 'https://i.imgur.com/rN7hY6om.jpg',
  alt: 'Una escultura negra mate donde los elementos individuales son inicialmente indistinguibles.'
}, {
  name: 'Aureole',
  artist: 'Ranjani Shettar',

>   description: 'SShettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad y una "fina síntesis de materiales inverosímiles.',

Reviewer 2: ```suggestion
  description: 'Shettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad, y una "fina síntesis de materiales inverosímiles.',
```
Razón: Corregido un typo (Doble S), y una coma para mejorar cohesión (mantiene el sujeto, que es la descripción de las obras).

>         by {sculpture.artist}

Reviewer 2: 1. Falta traducir la preposición
   * Frase: "by"
   * Sugerencia: "por"
   * Razón: Así mantenemos la traducción de strings al español de manera consistente (la idea es mantenerlo en inglés, o en español, pero no ambas).

>         by {sculpture.artist}

Reviewer 2: ```suggestion
        por {sculpture.artist}
```
Razón: Así mantenemos la traducción de strings al español de manera consistente (la idea es mantenerlo en inglés, o en español, pero no ambas).

}, {
  name: 'Eternal Presence',
  artist: 'John Woodrow Wilson',

>   description: 'Esta enorme flor plateada (75 pies o 23 m) se encuentra en Buenos Aires. Está diseñado para moverse, cerrando sus pétalos por la tarde o cuando soplan fuertes vientos y abriéndolos por la mañana.',

Reviewer 2: ```suggestion
  description: 'Wilson era conocido por su preocupación por la igualdad y la justicia social, así como por las cualidades esenciales y espirituales de la humanidad. Esta enorme pieza de bronce (de 7 pies o 2,13 metros) representa lo que él describió como "una presencia negra simbólica impregnada de un sentido de humanidad universal".',
```
Razón: Este par key-value está desfasado, la descripción corresponde a la key anterior (la description de la Floraris Genérica).

}, {
  name: 'Terracotta Army',
  artist: 'Artista Desconocido',
  description: 'El Ejército de terracota es una colección de esculturas de terracota que representan los ejércitos de Qin Shi Huang, el primer emperador de China. El ejército constaba de más de 8.000 soldados, 130 carros con 520 caballos y 150 caballos de caballería.',
  url: 'https://i.imgur.com/HMFmH6m.jpg',
  alt: '12 esculturas de terracota de guerreros solemnes, cada uno con una expresión facial y una armadura únicas.'
}, {
  name: 'Lunar Landscape',
  artist: 'Louise Nevelson',
  description: 'Nevelson era conocida por recoger objetos de los escombros de la ciudad de Nueva York, que luego ensamblaría en construcciones monumentales. En este, usó partes dispares como un poste de la cama, un alfiler de malabares y un fragmento de asiento, clavándolos y pegándolos en cajas que reflejan la influencia de la abstracción geométrica del espacio y la forma del cubismo.',
  url: 'https://i.imgur.com/rN7hY6om.jpg',
  alt: 'Una escultura negra mate donde los elementos individuales son inicialmente indistinguibles.'
}, {
  name: 'Aureole',
  artist: 'Ranjani Shettar',

>   description: 'SShettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad y una "fina síntesis de materiales inverosímiles.',

Reviewer 2: ```suggestion
  description: 'Shettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad, y una "fina síntesis de materiales inverosímiles.',
```
Razón: Corregido un typo (Doble S), y una coma para mejorar cohesión (mantiene el sujeto, que es la descripción de las obras).

</Sandpack>

Esto es lo que hace que el estado sea diferente de las variables regulares que declaramos en la parte superior de un módulo. El estado no está vinculado a una llamada de función en particular o a un lugar en el código, pero es "local" al lugar específico en la pantalla. Se han renderizado dos componentes `<Gallery />`, por lo que su estado se almacena por separado.

> También observemos cómo el componente de `Página` no "sabe" nada sobre el estado del componente `Galería` o incluso si tiene algún estado. A diferencia de las props, **el estado es totalmente privado para el componente que lo declara.** El componente padre no puede cambiarlo. Esto permite agregar el estado a cualquier componente o eliminarlo sin afectar al resto de los componentes.

Reviewer 2: 1. Mantener consistencia con el embed del CodeSandBox
   * Frase: "cómo el componente de `Página` no "sabe" nada sobre el estado del componente `Galería`"
   * Sugerencia: "cómo el componente de la página `Page` no "sabe" nada sobre el estado del componente de la galería `Galery`"
   * Razón: De tal manera podemos mantener la consistencia y hacer referencia a lo mismo tanto en la explicación como en el _CodeSandBox_ a lo largo del capítulo (En el código no fue traducido el nombre del componente).

> También observemos cómo el componente de `Página` no "sabe" nada sobre el estado del componente `Galería` o incluso si tiene algún estado. A diferencia de las props, **el estado es totalmente privado para el componente que lo declara.** El componente padre no puede cambiarlo. Esto permite agregar el estado a cualquier componente o eliminarlo sin afectar al resto de los componentes.

Reviewer 2: ```suggestion
También observemos cómo el componente de la página `Page` no "sabe" nada sobre el estado del componente de la galería `Galery`, ni siquiera si es que posee algún estado. A diferencia de las props, **el estado es totalmente privado para el componente que lo declara.** El componente padre no puede cambiarlo. Esto permite agregar el estado a cualquier componente o eliminarlo sin afectar al resto de los componentes.
```
Razón: De tal manera podemos mantener la consistencia y hacer referencia a lo mismo tanto en la explicación como en el _CodeSandBox_ a lo largo del capítulo (En el código no fue traducido el nombre del componente).

¿Qué pasaría si quisieramos que ambas galerías mantuvieran sus estados sincronizados? La forma correcta de hacerlo en React es *eliminar* el estado de los componentes secundarios y agregarlo a su padre más cercano. Las próximas páginas se centrarán en organizar el estado de un solo componente, pero volveremos a este tema en [Compartir estado entre componentes.](/learn/sharing-state-between-components)

<Recap>

* Debemos utilizar una variable de estado cuando necesitamos que un componente necesite "recordar" alguna información entre renderizaciones.
* Las variables de estado se declaran llamando al Hook `useState`.
* Los Hooks son funciones especiales que comienzan con `use`. Nos permiten "enlazarnos" a funciones de React como el estado.

> * Los Hooks se importan de la siguiente manera: deben llamarse incondicionalmente. Llamar Hooks, incluido `useState`, solo es válido en el nivel superior de un componente u otro Hook.

Reviewer 2: 1. Parafraseo
   * Frase: "Los Hooks se importan de la siguiente manera: deben llamarse incondicionalmente. Llamar Hooks, incluido `useState`, solo es válido en el nivel superior de un componente u otro Hook.""
   * Sugerencia: "Evita llamar a Hooks de manera anidada (por ejemplo, dentro de bucles, condicionales o declaraciones de funciones anidadas). Llamar a Hooks, incluyendo al `useState`, solo es válido en el nivel superior de un componente u otro Hook."
   * Razón: La traducción directa pierde algo de significado, por lo que podemos cambiar algunas palabras para mantenerlo en español.

> * Los Hooks se importan de la siguiente manera: deben llamarse incondicionalmente. Llamar Hooks, incluido `useState`, solo es válido en el nivel superior de un componente u otro Hook.

Reviewer 2: ```suggestion
*Evita llamar a Hooks de manera anidada (por ejemplo, dentro de bucles o condicionales). Llamar a Hooks -incluyendo al useState- solo es válido en el nivel superior de un componente u otro Hook.
```
Razón: La traducción directa pierde algo de significado, por lo que podemos cambiar algunas palabras para mantenerlo en español.

> * El Hook `useState` retorna un arreglo de dos valores: el estado actual y la función para actualizarlo.

Reviewer 2: 1. Convención de [traducción común](https://github.com/reactjs/es.reactjs.org/blob/main/TRANSLATION.md#traducciones-comunes)
   * Frase: "retorna un arreglo"
   * Sugerencia: "retorna un _array_"
   * Razón: Así mantenemos la consistencia con los usos anteriores y posteriores en este capítulo, y los capítulos anteriores en la documentación.

> * El Hook `useState` retorna un arreglo de dos valores: el estado actual y la función para actualizarlo.

Reviewer 2: ```suggestion
* El Hook `useState` retorna un *array* de dos valores: el estado actual y la función para actualizarlo.
```
Razón: Así mantenemos la consistencia con los usos anteriores y posteriores en este capítulo, y los capítulos anteriores en la documentación.

}, {
  name: 'Eternal Presence',
  artist: 'John Woodrow Wilson',

>   description: 'Esta enorme flor plateada (75 pies o 23 m) se encuentra en Buenos Aires. Está diseñado para moverse, cerrando sus pétalos por la tarde o cuando soplan fuertes vientos y abriéndolos por la mañana.',

Reviewer 2: ```suggestion
  description: 'Wilson era conocido por su preocupación por la igualdad y la justicia social, así como por las cualidades esenciales y espirituales de la humanidad. Esta enorme pieza de bronce (de 7 pies o 2,13 metros) representa lo que él describió como "una presencia negra simbólica impregnada de un sentido de humanidad universal".',
```
Razón: Este par key-value está desfasado, la descripción corresponde a la key anterior (la description de la Floraris Genérica).

}, {
  name: 'Terracotta Army',
  artist: 'Artista Desconocido',
  description: 'El Ejército de terracota es una colección de esculturas de terracota que representan los ejércitos de Qin Shi Huang, el primer emperador de China. El ejército constaba de más de 8.000 soldados, 130 carros con 520 caballos y 150 caballos de caballería.',
  url: 'https://i.imgur.com/HMFmH6m.jpg',
  alt: '12 esculturas de terracota de guerreros solemnes, cada uno con una expresión facial y una armadura únicas.'
}, {
  name: 'Lunar Landscape',
  artist: 'Louise Nevelson',
  description: 'Nevelson era conocida por recoger objetos de los escombros de la ciudad de Nueva York, que luego ensamblaría en construcciones monumentales. En este, usó partes dispares como un poste de la cama, un alfiler de malabares y un fragmento de asiento, clavándolos y pegándolos en cajas que reflejan la influencia de la abstracción geométrica del espacio y la forma del cubismo.',
  url: 'https://i.imgur.com/rN7hY6om.jpg',
  alt: 'Una escultura negra mate donde los elementos individuales son inicialmente indistinguibles.'
}, {
  name: 'Aureole',
  artist: 'Ranjani Shettar',

>   description: 'SShettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad y una "fina síntesis de materiales inverosímiles.',

Reviewer 2: ```suggestion
  description: 'Shettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad, y una "fina síntesis de materiales inverosímiles.',
```
Razón: Corregido un typo (Doble S), y una coma para mejorar cohesión (mantiene el sujeto, que es la descripción de las obras).

}, {
  name: 'Eternal Presence',
  artist: 'John Woodrow Wilson',

>   description: 'Esta enorme flor plateada (75 pies o 23 m) se encuentra en Buenos Aires. Está diseñado para moverse, cerrando sus pétalos por la tarde o cuando soplan fuertes vientos y abriéndolos por la mañana.',

Reviewer 2: ```suggestion
  description: 'Wilson era conocido por su preocupación por la igualdad y la justicia social, así como por las cualidades esenciales y espirituales de la humanidad. Esta enorme pieza de bronce (de 7 pies o 2,13 metros) representa lo que él describió como "una presencia negra simbólica impregnada de un sentido de humanidad universal".',
```
Razón: Este par key-value está desfasado, la descripción corresponde a la key anterior (la description de la Floraris Genérica).

}, {
  name: 'Terracotta Army',
  artist: 'Artista Desconocido',
  description: 'El Ejército de terracota es una colección de esculturas de terracota que representan los ejércitos de Qin Shi Huang, el primer emperador de China. El ejército constaba de más de 8.000 soldados, 130 carros con 520 caballos y 150 caballos de caballería.',
  url: 'https://i.imgur.com/HMFmH6m.jpg',
  alt: '12 esculturas de terracota de guerreros solemnes, cada uno con una expresión facial y una armadura únicas.'
}, {
  name: 'Lunar Landscape',
  artist: 'Louise Nevelson',
  description: 'Nevelson era conocida por recoger objetos de los escombros de la ciudad de Nueva York, que luego ensamblaría en construcciones monumentales. En este, usó partes dispares como un poste de la cama, un alfiler de malabares y un fragmento de asiento, clavándolos y pegándolos en cajas que reflejan la influencia de la abstracción geométrica del espacio y la forma del cubismo.',
  url: 'https://i.imgur.com/rN7hY6om.jpg',
  alt: 'Una escultura negra mate donde los elementos individuales son inicialmente indistinguibles.'
}, {
  name: 'Aureole',
  artist: 'Ranjani Shettar',

>   description: 'SShettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad y una "fina síntesis de materiales inverosímiles.',

Reviewer 2: ```suggestion
  description: 'Shettar fusiona lo tradicional y lo moderno, lo natural y lo industrial. Su arte se centra en la relación entre el hombre y la naturaleza. Su trabajo fue descrito como convincente tanto abstracta como figurativamente, desafiando la gravedad, y una "fina síntesis de materiales inverosímiles.',
```
Razón: Corregido un typo (Doble S), y una coma para mejorar cohesión (mantiene el sujeto, que es la descripción de las obras).
